src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/FallbackRazorConfiguration.cs

```diff
@@ -2,114 +2,58 @@
          RazorLanguageVersion.Version_3_0,
          "MVC-3.0",
-         new[] { new FallbackRazorExtension("MVC-3.0"), });
+         [new("MVC-3.0")]);
 
-    public static readonly RazorConfiguration MVC_5_0 = new FallbackRazorConfiguration(
+    public static readonly RazorConfiguration MVC_5_0 = new(
          RazorLanguageVersion.Version_5_0,
          // Razor 5.0 uses MVC 3.0 Razor configuration.
          "MVC-3.0",
-         new[] { new FallbackRazorExtension("MVC-3.0"), });
+         [new("MVC-3.0")]);
 
-    public static readonly RazorConfiguration Latest = new FallbackRazorConfiguration(
+    public static readonly RazorConfiguration Latest = new(
          RazorLanguageVersion.Latest,
          // Razor latest uses MVC 3.0 Razor configuration.
          "MVC-3.0",
-         new[] { new FallbackRazorExtension("MVC-3.0"), });
+         [new("MVC-3.0")]);
 
     public static RazorConfiguration SelectConfiguration(Version version)
-    {
-        if (version.Major == 1 && version.Minor == 0)
-        {
-            return MVC_1_0;
-        }
-        else if (version.Major == 1 && version.Minor == 1)
-        {
-            return MVC_1_1;
-        }
-        else if (version.Major == 2 && version.Minor == 0)
-        {
-            return MVC_2_0;
-        }
-        else if (version.Major == 2 && version.Minor >= 1)
-        {
-            return MVC_2_1;
-        }
-        else if (version.Major == 3 && version.Minor == 0)
-        {
-            return MVC_3_0;
-        }
-        else if (version.Major == 5 && version.Minor == 0)
-        {
-            return MVC_5_0;
-        }
-        else
-        {
-            return Latest;
-        }
-    }
-
-    public FallbackRazorConfiguration(
-        RazorLanguageVersion languageVersion,
-        string configurationName,
-        RazorExtension[] extensions,
-        bool useConsolidatedMvcViews = false)
-    {
-        if (languageVersion is null)
-        {
-            throw new ArgumentNullException(nameof(languageVersion));
-        }
-
-        if (configurationName is null)
-        {
-            throw new ArgumentNullException(nameof(configurationName));
-        }
-
-        if (extensions is null)
-        {
-            throw new ArgumentNullException(nameof(extensions));
-        }
-
-        LanguageVersion = languageVersion;
-        ConfigurationName = configurationName;
-        Extensions = extensions;
-        UseConsolidatedMvcViews = useConsolidatedMvcViews;
-    }
-
-    public override string ConfigurationName { get; }
-
-    public override IReadOnlyList<RazorExtension> Extensions { get; }
-
-    public override RazorLanguageVersion LanguageVersion { get; }
-
-    public override bool UseConsolidatedMvcViews { get; }
+        => version switch
+        {
+            { Major: 1, Minor: 0 } => MVC_1_0,
+            { Major: 1, Minor: 1 } => MVC_1_1,
+            { Major: 2, Minor: 0 } => MVC_2_0,
+            { Major: 2, Minor: >= 1 } => MVC_2_1,
```

Comment:
Yes, we can just be specific. I wouldn't worry about `StartsWith(...)` in `ProjectEngineFactory`. I don't know that using `StartsWith` is worth it. First, it's not accurate, since it'd match "MVC-2.DavidSmellsNiceToday". (:smile:) Second, I think it'd be fast enough. _That said_, it seems like it's probably just better if I add an internal map to the compiler (using a `FrozenDictionary`) that everything uses.

Reply:
Actually, I don't think we should touch this particular list or change the `>=`. This is a very different code path that everything else. It appears to be called using the version number of the "Microsoft.AspNetCore.Mvc.Razor.dll" assembly reference in the user's project. I wouldn't be surprised if there's a 2.2 version of that assembly that we just treat as MVC-2.1. Let's leave the ">=" to avoid breaking folks on ASP .NET Core 2.2